### PR TITLE
Allowing deleting block by block id

### PR DESCRIPTION
**Pull request by athul7744**
> Unable to delete a specific block using block index in delete method.
> BlockManager.removeBlock supports block index value to be passed.

- Passing block index from input to BlockManager.removeBlock function.

### CODE REVIEW

**Comment by neSpecc** [comment]
Resolves #1107

**Comment by neSpecc** [comment]
Please, add a not in a CHANGELOG.md under the 2.18 section

File: types/api/blocks.d.ts [hunk 1]
```diff
@@ -26,8 +26,9 @@ export interface Blocks {
 
   /**
    * Removes current Block
+   * @param {number} index
    */
-  delete(): void;
+  delete(index: number): void;
```
Comment (neSpecc) [inline types/api/blocks.d.ts:31]: maybe it would better to make this parameter optional?

Comment (athul7744) [reply types/api/blocks.d.ts:31]: So that it will work with currentBlockValue? Ok, I'll change it.

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 - `Improvements` - Deprecated TSLint replaced with ESLint, old config changed to [CodeX ESLint Config](https://github.com/codex-team/eslint-config).
 - `Improvements` - Adjusted GitHub action for ESLint.
+- `Improvements` - Allow deletion of blocks by specifying block index.
```
Comment (neSpecc) [inline docs/CHANGELOG.md:7]:
```suggestion
- `Improvements` - Blocks API: allow deletion of blocks by specifying block index via `blocks.delete()`.
```

File: src/components/modules/api/blocks.ts [hunk 3]
```diff
@@ -19,7 +19,7 @@ export default class BlocksAPI extends Module {
       clear: () => this.clear(),
       render: (data: OutputData) => this.render(data),
       renderFromHTML: (data: string) => this.renderFromHTML(data),
-      delete: () => this.delete(),
+      delete: (index: number) => this.delete(index),
```
Comment (neSpecc) [inline src/components/modules/api/blocks.ts:22]: still no optional here

Comment (athul7744) [reply src/components/modules/api/blocks.ts:22]: sorry about that, must've missed it 😄

File: types/api/blocks.d.ts [hunk 4]
```diff
@@ -26,8 +26,9 @@ export interface Blocks {
 
   /**
    * Removes current Block
+   * @param {number} index
```
Comment (neSpecc) [inline types/api/blocks.d.ts:29]: please describe a parameter

**Comment by athul7744** [comment]
Any other changes?

File: types/api/blocks.d.ts [hunk 5]
```diff
@@ -26,8 +26,9 @@ export interface Blocks {
 
   /**
    * Removes current Block
+   * @param {number} index - block to delete
```
Comment (neSpecc) [inline types/api/blocks.d.ts:29]:
```suggestion
   * @param {number} index -index of a block to delete
```

**Comment by athul7744** [comment]
Conflicts have been resolved.

**Pull request merged**